DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/annotated_allocation/src/annotated_alloc.cpp

```diff
@@ -0,0 +1,135 @@
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+#include "annotated_class_util.hpp"
+
+constexpr int kBL1 = 1;
+constexpr int kAlignment = 32;
+constexpr int kWidth = 256;
+
+// Create type alias for the annotated kernel arguments, so it can be
+// reused in the annotated memory allocation in the host code
+// Each annotated pointer is configured with a unique `buffer_location`,
+// resulting in three unique Avalon memory-mapped host interfaces.
+using annotated_arg_t= sycl::ext::oneapi::experimental::annotated_arg<
+    int *, decltype(sycl::ext::oneapi::experimental::properties{
+        sycl::ext::intel::experimental::buffer_location<kBL1>,
+        sycl::ext::intel::experimental::dwidth<kWidth>,
+        sycl::ext::oneapi::experimental::alignment<kAlignment> })>;
+
+struct VectorAdd {
+  annotated_arg_t a;
+  int size;
+
+  void operator()() const {
+    #pragma unroll 8
+    for (int i = 0; i < size; i++) {
+      a[i] *= 2;
+    }
+  }
+};
+
+bool checkResult(int *arr, int size) {
+  bool passed = true;
+  for (int i = 0; i < size; i++) {
+    int golden = 2 * i;
+      if (arr[i] != golden) {
+        std::cout << "ERROR! At index: " << i << " , expected: " << golden
+                  << " , found: " << arr[i] << "\n";
+        passed = false;
+      }
+    }
+    return passed;
+}
+
+bool runWithUsmMalloc(sycl::queue &q) {
+    // Create and initialize the host arrays
+    constexpr int kN = 8;
+    std::cout << "Elements in vector : " << kN << "\n";
```

Comment:
```suggestion
    std::cout << "using aligned_alloc_shared to allocate a block of shared memory\n";
```
print something that's actually interesting to the user :)